client/ayon_maya/plugins/load/load_layout.py

```diff
@@ -0,0 +1,223 @@
+from maya import cmds
+import math
+import json
+import collections
+import ayon_api
+from ayon_maya.api import plugin
+from ayon_maya.api.lib import unique_namespace
+from ayon_core.pipeline import (
+    load_container,
+    get_representation_path,
+    discover_loader_plugins,
+    loaders_from_representation,
+    get_current_project_name
+)
+from maya.api import OpenMaya as om
+from ayon_maya.api.pipeline import containerise
+
+
+class LayoutLoader(plugin.Loader):
+    """Layout Loader (json)"""
+
+    product_types = {"layout"}
+    representations = {"json"}
+
+    label = "Layout Loader"
+    order = -10
+    icon = "code-fork"
+    color = "orange"
+
+    def _get_repre_entities_by_version_id(self, data):
+        version_ids = {
+            element.get("version")
+            for element in data
+        }
+        version_ids.discard(None)
+
+        output = collections.defaultdict(list)
+        if not version_ids:
+            return output
+
+        project_name = get_current_project_name()
+        repre_entities = ayon_api.get_representations(
+            project_name,
+            representation_names={"fbx", "abc"},
+            version_ids=version_ids,
+            fields={"id", "versionId", "name"}
+        )
+        for repre_entity in repre_entities:
+            version_id = repre_entity["versionId"]
+            output[version_id].append(repre_entity)
+        return output
+
+    @staticmethod
+    def _get_loader(loaders, product_type):
+        name = ""
+        if product_type in {
+            "rig", "model", "camera",
+            "animation", "staticMesh",
+            "skeletalMesh"}:
+                name = "ReferenceLoader"
+
+        if name == "":
+            return None
+
+        for loader in loaders:
+            if loader.__name__ == name:
+                return loader
+
+        return None
+
+    def get_asset(self, containers, instance_name):
+        asset = None
+        containers = [con for con in containers if con.startswith(instance_name)]
+        for container in containers:
+            namespace = cmds.getAttr(f"{container}.namespace")
+            asset = [asset for asset in cmds.ls(f"{namespace}:*", assemblies=True)][0]
+        return asset
+
+    def _process_element(self, element, repre_entities_by_version_id):
+        repre_id = None
+        repr_format = None
+        version_id = element.get("version")
+        if version_id:
+            repre_entities = repre_entities_by_version_id[version_id]
+            if not repre_entities:
+                self.log.error(
+                    "No valid representation found for version"
+                    f" {version_id}")
+                return
+            # always use the first representation to load
+            # If reference is None, this element is skipped, as it cannot be
+            # imported in Maya, repre_entities must always be the first one
```

Comment:
I understand you added this comment based on my PR feedback - however, the comment does not explain **why** we do this? WHY is always using the first representation the 'good' choice? (What if for whatever reason the first representation is an image or thumbnail or whatever?)

Reply:
it is only applied for the layout published from maya and re-loaded back to maya, it errors out for the repre_id

Reply:
Which raises another question - why does that fail, etc? Like _what_ is the use case? And do we think that's completely fine? Are we lacking some data? This seems to be hiding a bug in plain sight? What am I missing?

Reply:
> Which raises another question - why does that fail, etc? Like _what_ is the use case? And do we think that's completely fine? Are we lacking some data? This seems to be hiding a bug in plain sight? What am I missing?

But it works when I loaded the layout from unreal, it works. Maybe I need to test it multiple times to ensure if there is anything missing.